src/settings/config.rs

```diff
@@ -122,18 +122,32 @@ impl TryFrom<&OwnedStruct> for ShellConfig {
     }
 }
 
+impl Default for UiConfig {
+    fn default() -> Self {
+        UiConfig {
+            auto_commit: true,
+            prompt: None,
+            format: FormatMode::default(),
+            edit_mode: EditMode::default(),
+            display_welcome: true,
+            display_ctrl_signals: true,
+            display_query_metrics: true,
+            terminator_required: false,
+        }
```

Comment:
I think it might be nicer to write this like

```rust
UiConfig {
    auto_commit: true,
    // and others
   .. Default::default()
}
```

This makes it clearer than you're using defaults where possible but doing some overrides.

Reply:
Agree, but seems the `..` syntax won't work here because it will recursively call the `default` function
```
UiConfig {
    auto_commit: true,
    display_welcome: true,
    display_ctrl_signals: true,
    display_query_metrics: true,
    .. Default::default()
}
```
We need to call the `default` function manually for each un-override field
```
UiConfig {
    auto_commit: true,
    display_welcome: true,
    display_ctrl_signals: true,
    display_query_metrics: true,
    terminator_required: Default::default(),
    prompt: Default::default(),
    format: Default::default(),
    edit_mode: Default::default(),
}
```